src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -456,7 +459,8 @@ private IParameterReferenceOperation CreateBoundParameterOperation(BoundParamete
             SyntaxNode syntax = boundParameter.Syntax;
             ITypeSymbol type = boundParameter.Type;
             Optional<object> constantValue = ConvertToOptional(boundParameter.ConstantValue);
-            bool isImplicit = boundParameter.WasCompilerGenerated;
+            // We want to mark compiler generated parameter reference for range variables as explicit because we don't expose any operation nodes for range variables.
+            bool isImplicit = boundParameter.WasCompilerGenerated && boundParameter.AssociatedRangeVariable == null;
```

Comment:
> && boundParameter.AssociatedRangeVariable == null [](start = 65, length = 50)

This doesn't feel right only top level node in the BoundRangeVariable.Value should be explicit and only if BoundRangeVariable itself wasnt compiler generated